mreg/api/permissions.py

```diff
@@ -134,9 +135,11 @@ def _deny_superuser_only_names(data=None, name=None, view=None, request=None):
             if 'host' in data:
                 name = data['host'].name
 
-    # Underscore is allowed for non-superuser in SRV records
+    # Underscore is allowed for non-superuser in SRV records,
+    # and for members of <DNS_UNDERSCORE_GROUP> in all records.
     if '_' in name and not isinstance(view, (mreg.api.v1.views.SrvDetail,
-                                             mreg.api.v1.views.SrvList)):
+                                             mreg.api.v1.views.SrvList)) \
+                   and not request_in_settings_group(request, DNS_UNDERSCORE_GROUP):
```

Comment:
This... works, and as the code base is now is probably the correct implementation, but checking permissions by either the class of calling view or if the request (hopefully request.user) is a member of a group? It looks really ugly in my opinion. However, its the way its done right now and we should probably do a larger cleanup pass rather than making piecemeal changes that only adds to the confusion.

Reply:
I agree. Let's take a look at how the permissions are implented at a later time. I think a re-write may be good, we can do things in a consistent way across the whole code base.